apollo-ios/Sources/Apollo/ExecutionSources/CacheDataExecutionSource.swift

```diff
@@ -67,6 +68,60 @@ struct CacheDataExecutionSource: GraphQLExecutionSource {
       }
     }
   }
+  
+  private func resolveCacheKey(
+    with info: FieldExecutionInfo,
+    on object: Record
+  ) throws -> AnyHashable? {
+    
+    let fieldTypename = typename(for: info.field)
+    
+    // Programmatic field policy checks
+    switch info.field.type {
```

Comment:
This code is messy and hard to read. 

- Programmatic and declarative resolution should be broken out into separate functions. 
- Function calls with multiple params that exceed our standard line length should be properly formatted with each arg on a new line. (We need to set up linter rules eventually... but we do have a `swift-format` config in the project)
- Lines 83-85 are repeated at line 93-95, DRY.
- Formatting of cache key string should be a helper function. It's repeated 6 times here.